lua/telescope/make_entry.lua

```diff
@@ -899,28 +899,52 @@ function make_entry.gen_from_autocommands(_)
   end
 end
 
-function make_entry.gen_from_git_status(_)
+local git_icon_defaults = {
+  added     = "+",
+  changed   = "~",
+  copied    = ">",
+  deleted   = "-",
+  renamed   = "➡",
+  unmerged  = "‡",
+  untracked = "?"
+}
+
+
+function make_entry.gen_from_git_status(opts)
+  opts = opts or {}
+
+  local col_width = ((opts.git_icons and opts.git_icons.added) and opts.git_icons.added:len() + 2) or 2
   local displayer = entry_display.create {
-  separator = " ",
+  separator = "",
   items = {
-      { width = 1},
-      { width = 1},
+      { width = col_width},
+      { width = col_width},
       { remaining = true },
+      -- { width = 70 },
     }
   }
 
   local make_display = function(entry)
-    local modified = "TelescopeResultsDiffChange"
-    local staged = "TelescopeResultsDiffAdd"
+    local icons = opts.git_icons or git_icon_defaults
```

Comment:
We should probably do something where we just create a new icons table that has a merging of the passed in icons and the defaults, if you only want to override one thing.

Also, maybe a list of the names somewhere and/or an example with pretty icons.